File: typescript/src/admin/reset.ts
import type { Request, Response } from "express";
import { config } from "../config.js";
import { ForbiddenError } from "../api/errors.js";
import { dropUsers } from "../db/queries/dropUsers.js";

export async function handlerReset(_: Request, res: Response) {
    if (config.api.platform !== "dev") {
        throw new ForbiddenError("Reset is only allowed in dev environment");
    }
    await dropUsers();
    config.api.fileServerHits = 0;
    res.status(200).json({ message: "System reset successful" });
    return;
}

File: golang/reset.go
package main

import (
	"net/http"
	"fmt"
)

func (cfg *apiConfig) handlerMetrics(w http.ResponseWriter, r *http.Request) {
	w.Header().Add("Content-Type", "text/html")
	w.WriteHeader(http.StatusOK)
	w.Write([]byte(fmt.Sprintf(`
<html>

<body>
	<h1>Welcome, Chirpy Admin</h1>
	<p>Chirpy has been visited %d times!</p>
</body>

</html>
	`, cfg.fileserverHits.Load())))
}

func (cfg *apiConfig) handlerReset(w http.ResponseWriter, r *http.Request){
	if cfg.platform != "dev" {
		w.WriteHeader(http.StatusForbidden)
		w.Write([]byte("Reset is only allowed in dev environment."))
		return
	}
	
	cfg.fileserverHits.Store(0)
	err := cfg.db.DeleteUsers(r.Context())
	if err != nil {
		respondWithError(w, 500, "Couldn't delete Users", err)
		return
	}
	w.WriteHeader(http.StatusOK)
	w.Write([]byte("Hits reset to 0 and database reset to initial state."))
	return
}